### PR TITLE
Draft updates

### PR DESCRIPTION
**Pull request by Spuds**
Some more draft updates ...  fixed some things, probably broke some others.  Main change is to work with the quick reply when its not using the sceditor but just the textbox.  Also added in some additional checks to help prevent background errors when things go wrong, which they will, trust me on this

### CODE REVIEW

File: Sources/Drafts.php [hunk 1]
```diff
@@ -552,11 +558,10 @@ function showProfileDrafts($memID, $draft_type = 0)
 		$_REQUEST['viewscount'] = 10;
 
 	// Get the count of applicable drafts on the boards they can (still) see ...
-	// @todo .. should we just let them see their drafts even if they have lost board access ?
 	$request = $smcFunc['db_query']('', '
 		SELECT COUNT(id_draft)
 		FROM {db_prefix}user_drafts AS ud
-			INNER JOIN {db_prefix}boards AS b ON (b.id_board = ud.id_board AND {query_see_board})
```
Comment (norv) [inline Sources/Drafts.php:559]: Question is, will they be able to post the finished draft now? I was thinking at the inclusion of a board selection widget, such as the post select mod, for these cases.
(don't shoot, I'll be back on this - interwebz and RL permitting.)

**Comment by Spuds** [comment]
Humm ... maybe an option to do that.  With this change if they create a draft and then loose the access to that board, they will at least still be able to see the draft they created (but should they try to post it they will get a permissions error.)  We can probably do more but it is just a draft :smiley:

**Comment by norv** [comment]
Yep, that's what I meant. They'll load the draft, and it looks ready to post, but the action of posting it expects a board, eh. That's what reminded me of the old SiNaN's board select mod, which allows the user to choose a[noher] board from the post form.
/ducks

**Comment by Spuds** [comment]
I'll go back and take a look at that, I think its a bit over kill for a draft but maybe .... I'll close this till I get done.

**Pull request closed without merging**